### ISSUE TITLE
Using a model with LeakyRelu in Tensorflow lite micro not supported?

### ISSUE THREAD

**Issue by RolfSchlup** [tensorflow/tensorflow#42465]
@tensorflow/micro

**System information**
- Host OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- TensorFlow installed from (source or binary):
- Tensorflow version (commit SHA if source):
- Target platform (e.g. Arm Mbed OS, Arduino Nano 33 etc.):

**Describe the problem**
I created a CNN model for analyzing a sensor grid that uses LeakyReLU.  Now I've been tasked to make the trained model work on a microcontroller.  The model loads fine ( after converting it etc. ), but as soon as I have it instatiate the micro interpreter it tells me that LeakyRelu is not supported mainly by saying "
Didn't find op for builtin opcode 'LEAKY_RELU' version '1'
Failed to get registration from op code LEAKY_RELU
"
**Please provide the exact sequence of commands/steps when you ran into the problem**

The following is the setup code I have so far

    static tflite::MicroErrorReporter micro_error_reporter;
    mErrorReporter = &micro_error_reporter;
    
    mModel = tflite::GetModel(model_kara_cnn_tflite);
    if (mModel->version() != TFLITE_SCHEMA_VERSION) {
      TF_LITE_REPORT_ERROR(mErrorReporter, "Model provided is schema version %d not equal to supported version %d.", mModel->version(), TFLITE_SCHEMA_VERSION);
      return false;
    }
    
    constexpr int tensor_arena_size = 16 * 1024;
    uint8_t tensor_arena[tensor_arena_size];
    tflite::MicroMutableOpResolver<5> resolver;
    resolver.AddConv2D();
    resolver.AddFullyConnected();
    resolver.AddMaxPool2D();
    resolver.AddSoftmax();
    resolver.AddBuiltin(tflite::BuiltinOperator_LEAKY_RELU, *tflite::ops::builtin::Register_LEAKY_RELU());

    tflite::MicroInterpreter interpreter(mModel, resolver, tensor_arena, tensor_arena_size, mErrorReporter);
    TfLiteStatus allocate_status = interpreter.AllocateTensors();
    if (allocate_status != kTfLiteOk) {
      TF_LITE_REPORT_ERROR(mErrorReporter, "AllocateTensors() failed");
      return false;
    }
    
    std::cout << "Setup complete" << std::endl;
    return true;

**Comment by RolfSchlup** [comment]
Changing the model to 
model = models.Sequential([
    layers.experimental.preprocessing.Rescaling(1./100, input_shape=(height, width, 1)),
    layers.Conv2D(32, kernel_size=(3, 3),activation='relu'),
    layers.Conv2D(64, (3, 3), activation='relu',padding='same'),
    layers.MaxPooling2D(pool_size=(2, 2),padding='same'),
    layers.Conv2D(128, (3, 3), activation='relu',padding='same'),
    layers.MaxPooling2D(pool_size=(2, 2),padding='same'),
    layers.Conv2D(64, 3, padding='same', activation='relu'),
    layers.MaxPooling2D(pool_size=(2, 2),padding='same'),
    layers.Flatten(),
    layers.Dense(128, activation='linear'),
    layers.Dense(num_classes, activation='softmax')
  ])

Gets me past the LeakyRelu problem, but now it says
"
Didn't find op for builtin opcode 'RESHAPE' version '1'
Failed to get registration from op code RESHAPE
"

**Comment by RolfSchlup** [comment]
So I can assume that LeakyReLU and the keras experimental preprocessing operations are not supported by tf lite micro?

**Comment by ymodak** [comment]
LeakyRelu is not supported in TF Lite Micro. The supported operations can be seen in the file [all_ops_resolver.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/micro/all_ops_resolver.cc).
Also see https://www.tensorflow.org/lite/microcontrollers#limitations to know more.

**Comment by adis300** [comment]
It would be great if LeakyReLU support is added.

**Comment by jeremy-u** [comment]
I know this is an old thread, but for other people who stumble across this.  LeakyRelu is not supported, but PRelu (parametric relu) is. It should be possible to create a leaky relu with your desired left-side slope by using PRelu and setting alpha_constraint and alpha_initializer appropriately.
